### PR TITLE
feat(content-blog): allow sorting posts in ascending order

### PR DESCRIPTION
**Pull request by cerkiewny**
## Motivation

Solves #5692

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Regression tests for older relevant tests.
Created new test that sorts posts in reverse creation order and compare it to default option.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

**Comment by Josh-Cena** [comment]
> Sure, sounds good I'll resubmit PR at the end or just after weekend

No need to close the current one, just change the current one

**Comment by cerkiewny** [comment]
> > Sure, sounds good I'll resubmit PR at the end or just after weekend
> 
> No need to close the current one, just change the current one
I meant I will resubmit this one :p

File: website/docs/api/plugins/plugin-content-blog.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ Accepted fields:
 | `feedOptions.description` | `string` | <code>\`${siteConfig.title} Blog\`</code> | Description of the feed. |
 | `feedOptions.copyright` | `string` | `undefined` | Copyright message. |
 | `feedOptions.language` | `string` (See [documentation](http://www.w3.org/TR/REC-html40/struct/dirlang.html#langcodes) for possible values) | `undefined` | Language metadata of the feed. |
+| `postSortDirection` | `string` | `'desc'` | One of 'desc', 'asc'. Governs the direction of blog post sorting. |
```
Comment (Josh-Cena) [inline website/docs/api/plugins/plugin-content-blog.md:62]:
```suggestion
| `postSortDirection` | <code>'asc' \| 'desc'</code> | `'desc'` | Governs the direction of blog post sorting. |
```

Maybe also rename to `'ascending' | 'descending'`? I think it's your C idiom kicking in again 😄

File: packages/docusaurus-plugin-content-blog/src/types.ts [hunk 2]
```diff
@@ -96,6 +96,7 @@ export type PluginOptions = RemarkAndRehypePluginOptions & {
   admonitions: Record<string, unknown>;
   authorsMapPath: string;
   readingTime: ReadingTimeFunctionOption;
+  postSortDirection: string;
```
Comment (Josh-Cena) [inline packages/docusaurus-plugin-content-blog/src/types.ts:99]:
```suggestion
  postSortDirection: 'asc' | 'desc';
```

Use union type to constrain to only two values.

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
I think it mostly looks good in terms of implementation. In terms of whether we want this API or not... not sure🤔

Several API design suggestions:

- `postSortDirection` => `sortPosts` (we can have other options in the future)
- `'desc'` => `'descending'` (it's better to be explicit IMO)

File: website/docs/api/plugins/plugin-content-blog.md [hunk 3]
```diff
@@ -59,6 +59,7 @@ Accepted fields:
 | `feedOptions.description` | `string` | <code>\`${siteConfig.title} Blog\`</code> | Description of the feed. |
 | `feedOptions.copyright` | `string` | `undefined` | Copyright message. |
 | `feedOptions.language` | `string` (See [documentation](http://www.w3.org/TR/REC-html40/struct/dirlang.html#langcodes) for possible values) | `undefined` | Language metadata of the feed. |
+| `postSorting` | <code>'descending' \| 'ascending' </code> | `'descending'` | Governs the direction of blog post sorting. |
```
Comment (Josh-Cena) [inline website/docs/api/plugins/plugin-content-blog.md:62]:
```suggestion
| `sortPosts` | <code>'descending' \| 'ascending' </code> | `'descending'` | Governs the direction of blog post sorting. |
```

**Review by Josh-Cena — APPROVED** [review]
LGTM. Looking for @slorber

**Comment by slorber** [comment]
LGTM

`sortPosts` is nice, gives the opportunity to eventually use a fn later if we change our mind

**Comment by slorber** [comment]
Just curious, are you using this feature @cerkiewny?

Do you expect the RSS feed to be in ascending order too? 🤔 

Asking for https://github.com/facebook/docusaurus/issues/9827#issuecomment-1936229920

**Pull request merged**